pkg/kube/snapshot/snapshot.go

```diff
@@ -62,7 +62,8 @@ type Snapshotter interface {
 	// 'waitForReady' will block the caller until the snapshot status is 'ReadyToUse'.
 	// or 'ctx.Done()' is signalled. Otherwise it will return immediately after the snapshot is cut.
 	// 'labels' can also be addded to the volume snapshot.
-	Create(ctx context.Context, name, namespace, pvcName string, snapshotClass *string, waitForReady bool, labels map[string]string) error
+	// 'annotations' can also be addded to the volume snapshot.
+	Create(ctx context.Context, name, namespace, pvcName string, snapshotClass *string, waitForReady bool, labels map[string]string, annotations map[string]string) error
```

Comment:
I was thinking if we should have a type called `SnapshotMeta` that should be passed to this method.
```
// has fields that can be set for created volume snapshot
type SnapshotMeta struct{
Name string
Namespace string
Labels map[string]string
Annotations map[string]string
}
```

Reply:
let's see what @PrasadG193 thinks.

Reply:
Sound good, we can use [meta/v1.ObjectMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta) as well

Reply:
@viveksinghggits @saima-s I was giving second thought on using `metav1.ObjectMeta`. It has a lot of other fields that we don't need. It might confuse consumer e.g they might set `OwnerReferences` in objectMeta but since we don't set it, it might lead to confusion. I think its better to declare a struct with only fields we allow to be set.

Reply:
Yes, I agree. But we can also inform about what field is to set or not in function definition as:
```
// snapshotMeta object has 'name' as name of the VolumeSnapshot.
// snapshotMeta object has 'namespace' as namespace of the PVC. VolumeSnapshot will be crated in the same namespace.
// snapshotMeta object has 'labels' that can also be addded to the volume snapshot.
// snapshotMeta object has 'annotations' that can also be addded to the volume snapshot.

```

Reply:
Yes, I also think it's better to not use `metav1.ObjectMeta` and have a new type/s like we discussed before.

Reply:
> Yes, I agree. But we can also inform about what field is to set or not in function definition as:

This just talks about the fields are the used, what about the other fields that are not used. We will have to mentioned that only these fields from `ObjectMeta` are used and other are ignored, which again, would confuse developers.

Reply:
Here, we can have 2 different struct for snapshot and snapshot content. Initially we will have similar set of fields as, later on we can separately include other fields if we want other to be configurable:
```
Name string
Namespace string
Labels map[string]string
Annotations map[string]string
```
Or we can have only 1 struct, as of now these 4 fields will suffice what we need. whdyt?

Reply:
I would prefer only 1 struct as I don't see any difference in the fields